Documentation/En/Index.xyl

```diff
@@ -3,8 +3,166 @@
+  <strong>task</strong>.</p>
+  <p>These tasks can be of several kinds such as sending a notification
+  (e.g. by email),  doing important operations on database, handling many files
+  etc. The idea is to <strong>unload</strong> the application from tasks that
+  could <strong>slow it down</strong>. These tasks may be run in
+  <strong>parallel</strong> and must not interfere with the execution of the
+  application.</p>
+  <p>Although the established protocol allows some <strong>integrity</strong>
+  and <strong>security</strong> for messages, it is preferable to do communication
+  from <strong>server to server</strong> and not through the client.
+  <p><code>Hoa\Worker</code> is made of two parts :</p>
+  <ul>
+    <li><code>Hoa\Worker\Backend\Shared</code> to
+    <strong>create</strong> a <strong>shared</strong> worker
+    between <strong>many</strong> processes ;</li>
+    <li><code>Hoa\Worker\Shared</code> to <strong>send</strong> messages to
+    the worker.</li>
+  </ul>
+
+  <h2 id="Shared_workers" for="main-toc">Shared workers</h2>
+
+  <p>A worker is primarily <strong>identified</strong> by
+  a <code>workerId</code>, sometime abbreviated <code>wid</code>. This identifier
+  is used to create a <strong>profile</strong> for our worker thanks to
+  <code>Hoa\Worker\Run</code> that does only this.
+  Only an identifier and the server address created in the worker are required
+  to establish a profile. Then, we can start a worker only from its identifier.</p>
+  <p>The first step will be to check that the profile exists through
+  <code>Hoa\Worker\Run::widExists</code> method. On the other hand, we will
+  create it with the <code>Hoa\Worker\Run::register</code> method. Thus :</p>
+  <pre><code class="language-php">if (false === Hoa\Worker\Run::widExists('demorker')) {
+    Hoa\Worker\Run::register('demorker', 'tcp://127.0.0.1:123456');
+}</code></pre>
+  <p>We are now sure that the profile exists. We can then <strong>start</strong>
+  our worker. For this, we must use its <strong>identifier</strong> and its
+  <strong>password</strong> (required to stop it) in the constructor of
+  <code>Hoa\Worker\Backend\Shared</code>. After creating the object, it will
+  broadcast only one category of <strong>event</strong> : <code>message</code>,
+  that we will listen to establish our own task processing <strong>protocol</strong>.
+  Finally the <code>Hoa\Worker\Backend\Shared::run</code> starts the worker :</p>
+  <pre><code class="language-php">$file   = new Hoa\File\Write(__DIR__ . DS . 'Log');
+$worker = new Hoa\Worker\Backend\Shared('demorker', 'iamapassword');
+$worker->on('message', function (Hoa\Core\Event\Bucket $bucket) use ($file) {
+    // compute messages.
+    $data = $bucket->getData();
+    $file->writeAll($data['message'] . "\n");
+});
+$worker->run();
+</code></pre>
+  <p>We are <strong>free</strong> to do what we want to process the
+  <strong>task</strong> !  The format is completely free : binary, compressed,
```

Comment:
```
- /strong> !
+ /strong>!
```
No space before an exclamation mark (same with question mark) in English :smiley:.